docs/content/how-to/aws/create-aws-hosted-cluster-autonode.md

```diff
@@ -0,0 +1,255 @@
+
+The following steps describes how to install OpenShift workload cluster
+on AWS with AutoNode feature by Karpenter.
+
+[ec2nodeclass]: https://karpenter.sh/docs/concepts/nodeclasses/
+[nodepools]: https://karpenter.sh/docs/concepts/nodepools/
+
+## Prerequisites
+
+- [Install the latest and greatest HyperShift CLI](https://hypershift-docs.netlify.app/getting-started/#prerequisites).
+    - Make sure all prerequisites have been satisfied (Pull Secret, Hosted Zone, OIDC Bucket, etc)
+- Enable AWS service-linked role for spot (one time by account where the hosted cluster will be installed):
+```sh
+aws iam create-service-linked-role --aws-service-name spot.amazonaws.com
+```
+
+### Create Karpenter IAM Role
+
+Create the IAM Role used by Karpenter controller.
+
+!!! warning "Temporary Step"
+    This is a temporary step while the issue [PODAUTO-302][PODAUTO-302]
+    is delivered.
+
+[PODAUTO-302]: https://issues.redhat.com/browse/PODAUTO-302
+
+```sh
+export ROLE_NAME=KarpenterNodeRole-agl
+aws iam create-role --role-name $ROLE_NAME --assume-role-policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}'
+
+aws iam attach-role-policy --role-name $ROLE_NAME --policy-arn arn:aws:iam::aws:policy/AdministratorAccess
+```
+
+### Export environment variables
+
+Export the following environment variables, adjusting according to your environment:
+
+```sh
+# AWS config
+export AWS_CREDS="$HOME/.aws/credentials"
+export AWS_DEFAULT_REGION=us-east-1
+
+# OpenShift credentials and configuration
+export CLUSTER_PREFIX=hcp-aws
+export CLUSTER_BASE_DOMAIN=devcluster.openshift.com
+export PULL_SECRET_FILE="${HOME}/.openshift/pull-secret-latest.json"
+export SSH_PUB_KEY_FILE=$HOME/.ssh/id_rsa.pub
+
+# Hypershift configuration
+## S3 Bucket name hosting the OIDC discovery documents
+export OIDC_BUCKET_NAME="${CLUSTER_PREFIX}-oidc"
+```
+
+### Install the Hypershift Operator
+
+Install controller enabling `tech-preview-no-upgrade` feature gate:
+
+```sh
+./hypershift install \
+  --oidc-storage-provider-s3-bucket-name="${OIDC_BUCKET_NAME}" \
+  --oidc-storage-provider-s3-credentials="${AWS_CREDS}" \
+  --oidc-storage-provider-s3-region="${AWS_DEFAULT_REGION}" \
+  --tech-preview-no-upgrade=true
+```
+
+- Check if controller is running as expected:
+
+```sh
+oc get all -n hypershift
+```
+
+## Create Workload Cluster with AutoNode
+
+Create the workload cluster with HyperShift AutoNode.
+
+Choose the desired target release image name ([release controller](https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/)).
+
+Create a hostead cluster, enabling the flag `--auto-node`:
+
+```sh
+HOSTED_CLUSTER_NAME=${CLUSTER_PREFIX}-419-ng
+OCP_RELEASE_IMAGE=registry.ci.openshift.org/ocp/release:4.19.0-0.nightly-2025-01-21-163021
+
+./hypershift create cluster aws \
```

Comment:
Should this note also be in the docs that @enxebre wrote in the Notes - HCP Karpenter Overview doc?
> Note: If you pass a release-image that’s not new enough, you need to pass --control-plane-operator-image  [quay.io/agarcial/cpo:autonode100](http://quay.io/agarcial/cpo:autonode100) to make sure the control plane operator has the latest changes (which includes tagging the data plane security group with the karpenter default ec2class discovery tag https://github.com/openshift/hypershift/blob/008f705cfc0732a80db8dad63d9ce749cb9377d8/karpenter-operator/controllers/karpenter/karpenter_controller.go#L284-L289).

Reply:
Good call. I would  defer to @enxebre .

When using the flag `--control-plane-operator-image`, it would be nice referencing on how to build or extract it (from release image? or hypershift registry?). Is there a doc fragment with that information?

I found only [this][this] reference:

https://github.com/openshift/hypershift/blob/6889612230b4caedddd1feb28fc783b72f03d99f/docs/content/how-to/disconnected/known-issues.md?plain=1#L121-L123

[this]: https://hypershift-docs.netlify.app/how-to/disconnected/known-issues/#hcp-hypershift-operator-on-disconnected-clusters-ignores-registryoverrides-inspecting-the-control-plane-operator-image

Reply:
Hi @enxebre do you think we need to keep the flag `--control-plane-operator-image` in this page?